src/main/java/com/prgrms/nabmart/domain/coupon/UserCoupon.java

```diff
@@ -0,0 +1,43 @@
+package com.prgrms.nabmart.domain.coupon;
+
+import com.prgrms.nabmart.domain.user.User;
+import com.prgrms.nabmart.global.BaseTimeEntity;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@Entity
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class UserCoupon extends BaseTimeEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long userCouponId;
+
+    @Column(nullable = false)
+    private boolean isUsed;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id")
+    private User user;
+
+    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.REMOVE)
+    @JoinColumn(name = "coupon_id")
+    private Coupon coupon;
+
+    public UserCoupon(User user, Coupon coupon) {
```

Comment:
```suggestion
    public UserCoupon(final User user, final Coupon coupon) {
```
추가로 User, Coupon null 검증하면 좋을 것 같아요!

Reply:
이 부분은 멘토님 의견을 통해 고려해봐요!!